### PR TITLE
Assignments dashboard

### LINKED ISSUE #2: Create Dashboard Page
- (for student) create a dashboard page containing the assignments to do and a table recording the past submissions
- (for tutor) create a dashboard page containing the assignment released, and a table recording the students related to the assignment, with their state of submission (maybe just SUBMITTED/NOT)

### PR DESCRIPTION
**Pull request by vivienherq**
# Pull Request

## Description

Created dashboard to show assignments for tutors and students (same view), as well as different views for submissions for tutors and students (currently switch between them by toggling).

## Related Issue(s)

Closes #2

## Screenshots (if applicable)

/assignments/dashboard
Screenshot 2024-03-17 at 3 23 18 PM

/assignments/submissions
Student's view
Screenshot 2024-03-17 at 3 23 43 PM
Tutor's view
Screenshot 2024-03-17 at 3 58 28 PM

## Checklist

- [ ] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [ ] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [ ] I will take care of the merging and delete the side-branch after the PR is merged.

## To Do
- [ ] add grades column (allow grades to be shown for final submission)
- [ ] highlight final (last) submission
- [x] tutor view for submissions page (show student submission details for each assignment)
- [x] styling

### CODE REVIEW

**Comment by tryyang2001** [comment]
> Should it show all the submissions for each assignment, or only the final submission for each assignment?

I think for now it is sufficient to just show the latest submission, unless it is easy to extend to each submission (since I thought to keep a record of all submissions will be the duty of History Service if we are building it)

> For students, would it be better to show the submissions on a dedicated "Submissions" page?

Sure, this works as well. I will leave it to you to design :)

> since I can't add a button to navigate to the assignment page within the accordion header

Ok, I didn't know about this. How about we add a link for the accordion header? So if you click the text it will navigate you to the assignment page, and if you only click the blank region then it will expand the accordion.

**Comment by vivienherq** [comment]
@junhonglow the issue with adding a link to the accordion header is that it will render the accordion itself useless, since clicking it will no longer open the accordion but navigate to another page... i will think about how else i can design this

**Comment by tryyang2001** [comment]
@vivienherq I think you tag the wrong person xD

Oh I was suggesting we only add a link to the text, not to the entire accordion header. But I think this is not crucial to have a link/button in the header, we can always let the user expand it before we provide the button/anything.

**Comment by junhonglow** [comment]
Works well functionally.

Although would it be possible to reduce some code repetition in the submissions page between the student and tutor view, maybe a check on whether the third table cell is required rather than the entirety of the component, since they share the last two columns.

File: frontend/src/app/(pages)/assignments/submissions/page.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                color="default"
+                selectionMode="single"
+                defaultSelectedKeys={[0]}
+                aria-label={`Submissions table for ${item.title}`}
+                shadow="none"
+              >
+                <TableHeader>
+                  <TableColumn>Name</TableColumn>
+                  <TableColumn>Submission Date and Time</TableColumn>
+                  <TableColumn width="20" align="end">
```
Comment (junhonglow) [inline frontend/src/app/(pages)/assignments/submissions/page.tsx:175]: I think we might want to standardize to tailwind css.

**Review by junhonglow — CHANGES_REQUESTED** [review]

**Comment by vivienherq** [comment]
> Works well functionally.
> 
> Although would it be possible to reduce some code repetition in the submissions page between the student and tutor view, maybe a check on whether the third table cell is required rather than the entirety of the component, since they share the last two columns.

I gave conditionally rendering the third column a try, but it results in an error that I can't seem to find the source of, but I will try again in future commits.

Screenshot 2024-03-18 at 11 11 25 AM

File: frontend/src/app/(pages)/assignments/submissions/page.tsx (hunk 1 above)
Comment (vivienherq) [reply frontend/src/app/(pages)/assignments/submissions/page.tsx:175]: Changed to tailwind where I can!

**Review by tryyang2001 — APPROVED** [review]
Tested on my local comp. LGTM, thanks for the effort!

**Pull request merged**